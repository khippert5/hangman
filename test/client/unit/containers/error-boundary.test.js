import React from 'react';
import { mount, shallow } from 'enzyme';
import mockConsole from 'jest-mock-console';

import ErrorBoundary from '../../../../src/client/containers/error-boundary';
import SystemErrorMessage from '../../../../src/client/components/system-error';

describe('Test application render', () => {
  let wrapper;
  beforeEach(() => {
    mockConsole(); // Restore the console
  });

  afterEach(() => {
    mockConsole(); // Restore the console
  });

  it('Successful render', async (done) => {
    wrapper = shallow(<ErrorBoundary />);
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('Catches an error', async (done) => {
    const TestComponent = () => { throw new Error('Error found'); }
    wrapper = mount(<ErrorBoundary>
      <TestComponent />
    </ErrorBoundary>);
    expect(wrapper.find(SystemErrorMessage).length).toEqual(1);
    done();
  });

});
