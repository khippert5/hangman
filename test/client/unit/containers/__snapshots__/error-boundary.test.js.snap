// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test application render Successful render 1`] = `""`;
